api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -183,44 +183,126 @@ message JwtHeader {
+  repeated JwtRequirement requires = 2;
+}
+
 // This is the Envoy HTTP filter config for JWT authentication.
-// [#not-implemented-hide:]
+//
+// For example:
+//
+// .. code-block:: yaml
+//
+//    - providers:
+//        issuer: issuer1
+//        audiences:
+//        - audience1
+//          audience2
+//        remote_jwks:
+//          http_uri:
+//            uri: https://example.com/.well-known/jwks.json
+//            cluster: example_jwks_cluster
+//    - providers:
+//        issuer: issuer2
+//        audiences:
+//        local_jwks:
+//          inline_string: jwks_string
+//    # Not jwt verification is required for /health path
+//    - rules:
+//        match:
+//          prefix: "/health"
+//    # Jwt verification for issuer1 is required for path prefixed with "prefix"
+//    - rules:
+//        match:
+//          prefix: "/prefix"
+//        - requires:
+//            issuer: issuer1
+//            audiences: [audience_override]
+//    # Jwt verification for issuer2 is required for all other requests.
+//    - rules:
+//        match:
+//          prefix: "/"
+//        - requires:
+//            issuer: issuer2
+//
+//// [#not-implemented-hide:]
 message JwtAuthentication {
-  // List of JWT rules to valide.
-  repeated JwtRule rules = 1;
+  // List of JWT providers.
+  repeated JwtProvider providers = 1;
 
+  // DEPRECATED: Please use "routes" field. if "routes" field is not empty,
+  // this field will not be used.
   // If true, the request is allowed if JWT is missing or JWT verification fails.
   // Default is false, a request without JWT or failed JWT verification is not allowed.
   bool allow_missing_or_failed = 2;
 
-  // This field lists the patterns allowed to bypass JWT verification. This only applies when
-  // `allow_missing_or_failed_jwt` is false. Under this config, if a request doesn't have JWT, it
-  // will be rejected. But some requests still needed to be forwarded without JWT, such as OPTIONS
-  // for CORS and some health checking paths.
-  //
-  // Examples: bypass all CORS options requests
-  //
-  // .. code-block:: yaml
-  //
-  //   bypass:
-  //   - headers:
-  //     - name: :method
-  //       value: OPTIONS
-  //   - headers:
-  //     - name: :path
-  //       regex_match: /.*
+  // Specifies requirements based on the route matches. The first matched requirement will be
+  // applied. If there are overlapped match conditions, please put the most specific match first.
+  // If PerRouteConfig is specified for the selected route, this field will not be uesd.
   //
-  // Examples: bypass /healthz check
+  // Examples
   //
   // .. code-block:: yaml
   //
-  //   bypass:
-  //   - headers:
-  //     - name: :method
-  //       value: GET
-  //   - headers:
-  //     - name: :path
-  //       exact_match: /healthz
+  //    - rules:
+  //      match:
+  //         prefix: "/healthz"
+  //    - rules:
+  //      match:
+  //         prefix: "/"
+  //      - requires:
+  //         issuer: issuer1
```

Comment:
Again, check the YAML, I think both match and requires need some tweaks.

Reply:
Done